src/components/AddOrganizationInEcosystem.tsx

```
@@ -0,0 +1,373 @@
+import { Roles } from '../utils/enums/roles';
+import { Button } from 'flowbite-react';
+import { addOrganizationInEcosystem } from '../api/ecosystem';
+import { pathRoutes } from '../config/pathRoutes';
+
+
+const initialPageState = {
+	page: 1,
+	search: '',
+	sortingOrder: 'desc',
+	pageSize: 3,
+	total: 100,
+	role: Roles.OWNER
+};
+
+interface IErrorOrg {
+	id: string;
+	error: string;
+}
+
+interface IErrorResponse {
+	statusCode: number;
+	message: string;
+	data?: {
+		orgId: string;
+	}
+	error?: string;
+}
+
+interface ICurrentPage {
+	page: number;
+	pageSize: number;
+	search: string;
+	role: string;
+	total: number;
+}
+
+const AddOrganizationInEcosystem = () => {
+	const [listAPIParameter, setListAPIParameter] = useState<ICurrentPage>(initialPageState);
+	const [errorList, setErrorList] = useState<IErrorOrg[]>([]);
+	const [loading, setLoading] = useState<boolean>(true);
+	const [error, setError] = useState<string | null>(null);
+	const [success, setSuccess] = useState<string | null>(null);
+	const [localOrgs, setLocalOrgs] = useState<string[]>([]);
+	const [pageInfo, setPageInfo] = useState({
+		totalItem: 0,
+		nextPage: 0,
+		lastPage: 0,
+	});
+	const [totalPages, setTotalPages] = useState<number>(0);
+	const [loader, setLoader] = useState(false);
+	const [organizationsList, setOrganizationsList] = useState<Array<Organisation> | null>(null);
+	const [tableData, setTableData] = useState<TableData[]>([])
+
+	const selectOrganization = async (item: Organisation, checked: boolean) => {
+		try {
+			const index = localOrgs?.length > 0 ? localOrgs.findIndex(ele => ele === item.id) : -1
+
+			if (index === -1) {
+				setLocalOrgs((prev: string[]) => [...prev, item.id])
+			} else {
+				const updateLocalOrgs = [...localOrgs]
+				if (!checked) {
+					updateLocalOrgs.splice(index, 1);
+				}
+				setLocalOrgs(updateLocalOrgs)
+			}
+		} catch (error) {
+			console.error("SELECTED ORGANIZATION:::", error)
```

We can throw an exception instead of adding a console log

---

Resolved